bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/plugin/ClangdPlugin.java

```diff
@@ -53,6 +54,7 @@ public void start(BundleContext context) throws Exception {
 		compileCommandsMonitor = new CompileCommandsMonitor(workspace).start();
 		cProjectChangeMonitor = new CProjectChangeMonitor().start();
 		configFileMonitor = new ClangdConfigFileMonitor(workspace).start();
+		new ClangdUtils().checkWorkspaceClangFormatFile(workspace);
```

Comment:
Synchronous IO operations during bundle activation is anti-pattern, since it may slow down the overall startup process.

Reply:
Okay, may be this file can be created when the LS gets started.

Reply:
...or better: prior to the start of the LS

Reply:
The `.clang-format` file will be created in a project, prior to  the opening of a file in that project.

Reply:
...if it's not yet exisiting